# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTextureUpload.cpp
# Commit: adc841b3e204
# Full Hash: adc841b3e204564ee8515a87206eb317275844c4
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-02-21 09:51:10
# Regressor Bug: 1616715
# File Overlap Count: 1
# Description:
#   Bug 1616715 - SurfaceFromElement gets alpha Premult unless opt-in to NonPremult. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D63421
# ==============================================================================

diff -r 7917b3a4a9a7 -r adc841b3e204 dom/canvas/WebGLTextureUpload.cpp
--- a/dom/canvas/WebGLTextureUpload.cpp	Fri Feb 21 03:11:23 2020 +0000
+++ b/dom/canvas/WebGLTextureUpload.cpp	Fri Feb 21 02:27:08 2020 +0000
@@ -170,14 +170,12 @@
   // same as drawImage.
   uint32_t flags = nsLayoutUtils::SFE_WANT_FIRST_FRAME_IF_IMAGE |
                    nsLayoutUtils::SFE_WANT_IMAGE_SURFACE |
-                   nsLayoutUtils::SFE_USE_ELEMENT_SIZE_IF_VECTOR;
+                   nsLayoutUtils::SFE_USE_ELEMENT_SIZE_IF_VECTOR |
+                   nsLayoutUtils::SFE_ALLOW_NON_PREMULT;
 
   if (mPixelStore.mColorspaceConversion == LOCAL_GL_NONE)
     flags |= nsLayoutUtils::SFE_NO_COLORSPACE_CONVERSION;
 
-  if (!mPixelStore.mPremultiplyAlpha)
-    flags |= nsLayoutUtils::SFE_PREFER_NO_PREMULTIPLY_ALPHA;
-
   RefPtr<gfx::DrawTarget> idealDrawTarget = nullptr;  // Don't care for now.
   auto sfer = nsLayoutUtils::SurfaceFromElement(
       const_cast<dom::Element*>(&elem), flags, idealDrawTarget);